README.md

```diff
@@ -659,10 +659,290 @@ To begin coding the basic execution functionality of our VM, we only need these
+
 ```go
-    //TODO
+func (vm *VirtualMachine) ComputeOperands(instruction Instruction) (Operands, error) {
+	var res *memory.MaybeRelocatable
+
+	dst_addr, err := vm.RunContext.ComputeDstAddr(instruction)
+	if err != nil {
+		return Operands{}, errors.New("FailedToComputeDstAddr")
+	}
+	dst, _ := vm.Segments.Memory.Get(dst_addr)
+
+	op0_addr, err := vm.RunContext.ComputeOp0Addr(instruction)
+	if err != nil {
+		return Operands{}, fmt.Errorf("FailedToComputeOp0Addr: %s", err)
+	}
+	op0, _ := vm.Segments.Memory.Get(op0_addr)
+
+	op1_addr, err := vm.RunContext.ComputeOp1Addr(instruction, op0)
+	if err != nil {
+		return Operands{}, fmt.Errorf("FailedToComputeOp1Addr: %s", err)
+	}
+	op1, _ := vm.Segments.Memory.Get(op1_addr)
+
+	if op0 == nil {
+		deducedOp0, deducedRes, err := vm.DeduceOp0(&instruction, dst, op1)
+		if err != nil {
+			return Operands{}, err
+		}
+		op0 = deducedOp0
+		if op0 != nil {
+			vm.Segments.Memory.Insert(op0_addr, op0)
+		}
+		res = deducedRes
+	}
+
+	if op1 == nil {
+		deducedOp1, deducedRes, err := vm.DeduceOp1(instruction, dst, op0)
+		if err != nil {
+			return Operands{}, err
+		}
+		op1 = deducedOp1
+		if op1 != nil {
+			vm.Segments.Memory.Insert(op1_addr, op1)
+		}
+		if res == nil {
+			res = deducedRes
+		}
+	}
+
+	if res == nil {
+		res, err = vm.ComputeRes(instruction, *op0, *op1)
+
+		if err != nil {
+			return Operands{}, err
+		}
+	}
+
+	if dst == nil {
+		deducedDst := vm.DeduceDst(instruction, res)
+		dst = deducedDst
+		if dst != nil {
+			vm.Segments.Memory.Insert(dst_addr, dst)
+		}
+	}
+
+	operands := Operands{
+		Dst: *dst,
+		Op0: *op0,
+		Op1: *op1,
+		Res: res,
+	}
+	return operands, nil
+}
+```
+
+##### ComputeDstAddr
+The method `ComputeDstAddr` computes the address of value that will be stored in the Destination operand. It checks which register is the relative to (wether AP or FP) and gets the direction by adding the instruction's first offset(off0) to the corresponding register.
+
+```go
+func (run_context RunContext) ComputeDstAddr(instruction Instruction) (memory.Relocatable, error) {
+	var base_addr memory.Relocatable
+	switch instruction.DstReg {
+	case AP:
+		base_addr = run_context.Ap
+	case FP:
+		base_addr = run_context.Fp
+	}
+
+	if instruction.Off0 < 0 {
+		return base_addr.SubUint(uint(math.Abs(float64(instruction.Off0))))
+	} else {
+		return base_addr.AddUint(uint(instruction.Off0))
+	}
+
+}
 ```
+###### ComputeOp0Addr
 
+The process is similar to compute the dst address.
+
+```go
+func (run_context RunContext) ComputeOp0Addr(instruction Instruction) (memory.Relocatable, error) {
+	var base_addr memory.Relocatable
+	switch instruction.Op0Reg {
+	case AP:
+		base_addr = run_context.Ap
+	case FP:
+		base_addr = run_context.Fp
+	}
+
+	if instruction.Off1 < 0 {
+		return base_addr.SubUint(uint(math.Abs(float64(instruction.Off1))))
+	} else {
+		return base_addr.AddUint(uint(instruction.Off1))
+	}
+}
+
+```
+
+###### ComputeOp1Addr
```

Comment:
Is `ComputeOp1Addr` included in the `ComputeDstAddr` section? If not, it should have 5 `#`, not 6.